api/v1beta1/hostedcluster_types.go

```diff
@@ -124,6 +124,27 @@ const (
 	// For example, to override the memory and cpu request for the Kubernetes APIServer:
 	// resource-request-override.hypershift.openshift.io/kube-apiserver.kube-apiserver: memory=3Gi,cpu=2000m
 	ResourceRequestOverrideAnnotationPrefix = "resource-request-override.hypershift.openshift.io"
+
+	// KubeVirtBaseDomainPassthroughAnnotation toggles whether or not an automatically generated base domain for
+	// the guest cluster should be used that is a subdomain of the management cluster's *.apps DNS.
+	//
+	// For the KubeVirt platform, the basedomain can be autogenerated using
+	// the *.apps domain of the management/infra hosting cluster
+	// This makes the guest cluster's base domain a subdomain of the
+	// hypershift infra/mgmt cluster's base domain.
+	//
+	// Example:
+	//   Infra/Mgmt cluster's DNS
+	//     Base: example.com
+	//     Cluster: mgmt-cluster.example.com
+	//     Apps:    *.apps.mgmt-cluster.example.com
+	//   KubeVirt Guest cluster's DNS
+	//     Base: apps.mgmt-cluster.example.com
+	//     Cluster: guest.apps.mgmt-cluster.example.com
+	//     Apps: *.apps.guest.apps.mgmt-cluster.example.com
+	//
+	// This is possible using OCP wildcard routes
```

Comment:
PR desc mention an API field but the PR implements an annotation.
Annotations used as in this PR are just opaque APIs that still need to follow support policies once we ship. Because of this I'm +1 to make this a field.